modules/kms-keyring-node/src/kms_hkeyring_node_helpers.ts

```diff
@@ -0,0 +1,617 @@
+    //# SHA-384 hash of the following byte strings, in the order listed:
+
+    //= aws-encryption-sdk-specification/framework/aws-kms/aws-kms-hierarchical-keyring.md#encryption-materials
+    //# All the above fields must be separated by a single NULL_BYTE `0x00`.
+    //#
+    //# | Field                  | Length (bytes) | Interpreted as      |
+    //# | ---------------------- | -------------- | ------------------- |
+    //# | Resource ID            | 1              | bytes               |
+    //# | Null Byte              | 1              | `0x00`              |
+    //# | Scope ID               | 1              | bytes               |
+    //# | Null Byte              | 1              | `0x00`              |
+    //# | Partition ID           | Variable       | bytes               |
+    //# | Null Byte              | 1              | `0x00`              |
+    //# | Logical Key Store Name | Variable       | UTF-8 Encoded Bytes |
+    //# | Null Byte              | 1              | `0x00`              |
+    //# | Branch Key ID          | Variable       | UTF-8 Encoded Bytes |
+
+    entryInfo = Buffer.concat([
+      //= aws-encryption-sdk-specification/framework/aws-kms/aws-kms-hierarchical-keyring.md#encryption-materials
+      //# - MUST be the Resource ID for the Hierarchical Keyring (0x02)
+      RESOURCE_ID,
+      NULL_BYTE,
+      //= aws-encryption-sdk-specification/framework/aws-kms/aws-kms-hierarchical-keyring.md#encryption-materials
+      //# - MUST be the Scope ID for Encrypt (0x01)
+      ENCRYPTION_SCOPE,
+      NULL_BYTE,
+      //= aws-encryption-sdk-specification/framework/aws-kms/aws-kms-hierarchical-keyring.md#encryption-materials
+      //# - MUST be the Partition ID for the Hierarchical Keyring
+      partitionId,
+      NULL_BYTE,
+      //= aws-encryption-sdk-specification/framework/aws-kms/aws-kms-hierarchical-keyring.md#encryption-materials
+      //# - MUST be the UTF8 encoded Logical Key Store Name of the keystore for the Hierarchical Keyring
+      logicalKeyStoreName,
+      NULL_BYTE,
+      //= aws-encryption-sdk-specification/framework/aws-kms/aws-kms-hierarchical-keyring.md#encryption-materials
+      //# - MUST be the UTF8 encoded branch-key-id
+      branchKeyIdAsBytes,
+    ])
+  }
+
+  // const entryInfo = versionAsBytes
+  //   ? Buffer.concat([
+  //       RESOURCE_ID,
+  //       NULL_BYTE,
+  //       DECRYPTION_SCOPE,
+  //       NULL_BYTE,
+  //       partitionId,
+  //       NULL_BYTE,
+  //       logicalKeyStoreName,
+  //       NULL_BYTE,
+  //       branchKeyIdAsBytes,
+  //       NULL_BYTE,
+  //       versionAsBytes,
+  //     ])
+  //   : Buffer.concat([
+  //       RESOURCE_ID,
+  //       NULL_BYTE,
+  //       ENCRYPTION_SCOPE,
+  //       NULL_BYTE,
+  //       partitionId,
+  //       NULL_BYTE,
+  //       logicalKeyStoreName,
+  //       NULL_BYTE,
+  //       branchKeyIdAsBytes,
+  //     ])
+
+  // encrypt the branch key id buffer with sha512
```

Comment:
*hiss* hashing is not encryption 
```suggestion
  // hash the branch key id buffer with sha512
```